lib/resources/auth/token_repository.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:synchronized/synchronized.dart';
+
+import 'package:wastexchange_mobile/util/jwt_utils.dart';
+
+/// Abstract implementation of TokenRepository.
+///
+/// TokenRepository can be used as base class for implementing the Authentication Token Functionality.
+abstract class TokenRepository {
```

Comment:
This abstract class could be 'BaseTokenRepository'

Reply:
Removed the base class as we decided to only use one type of token handling in this project.